docs/usage/filters-migration.md

```diff
@@ -0,0 +1,156 @@
+# Filters format migration done in 5.12
+
+## Context: why this migration
+
+Before 5.12, it was not possible to do some complex filters combinations: we couldn't imbricate filters with differents modes (and/or), filter on any available attribute or relation for a given entity type, or test empty fields.
+
+The former format was not adapted for such complex filters, and a refacto was necessary.
+Indeed:
+- the filters frontend and backend formats were different, requiring conversions,
+- the filter keys were enums, i.e. the keys should belong to the static list of the available keys for the given entity type,
+- in the frontend format, the operator was contained in the key, limiting the operators combinations and creating useless complexity of parsing,
+- the frontend format didn't contained a mode attribute, avoiding modes (and/or) customization,
+- the flat structure didn't enable filters imbrication at several levels.
+
+```
+// old filters format
+FrontendFilters: Map<string, { id: string | null, value: string }>
+
+BackendFilters {
+    key: string, // an enum that should be in the list of the available filter keys for given the entity type
+    values: string[],
+    operator: string,
+    filterMode: 'and' | 'or',
+}
+```
+
+Because changing filters format impacts almost everything in the platform, we decided to do a complete refacto once and for all, a refacto that would enable almost all complex combinations and compatible with our long term vision of filters in OpenCTI.
+
+## Content: what has been done
+
+The filters refacto bring major changes in the way filtering is done.
+
+- We change the filters formats (see FilterGroup type above):
+    - In the frontend, an operator and a mode are stored for each key,
+    - The new format enables filters imbrication thanks to the new attribute 'filterGroups',
+    - The keys are of type string (no more static list of enums).
+    - The 'values' attribute can no longer contain null values (use the 'nil' operator instead).
+
+- We also rename some filter keys, to be consistant with the entities schema definitions.
+
+- We implemented the handling of the different operators and modes in the backend.
+
+- We introduced a new operator: 'nil' and 'not_nil', enabling to test wether an attribute is empty or not.
+
+## Warnings: what you should do to ensure a correct migration
+
+We wrote a migration to convert all the old stored filters (filters contained in streams, taxii collections, feeds, triggers, workspaces) in the new format.
+But you might have things to change for your own bits of code (your own connectors, queries, python scripts...). For each filter you hardcoded, you should change the filters format in the new one.
```

Comment:
Question: is it really a 'you should change' and not a 'you must change before updating the platform to 5.12' ? If hardcoded filter prio to v5.12 are kept in custom code then it's not going to work anymore right ?

Reply:
right it's a must :) i'm going to change this :)